--- a/apps/konsole/src/main.cpp
+++ b/apps/konsole/src/main.cpp
@@ -137,6 +137,7 @@ void fillCommandLineOptions(KCmdLineOpti
     options.add("noclose",ki18n("Do not close the initial session automatically when it ends."));
     // TODO - Document this option more clearly
     options.add("p <property=value>",ki18n("Change the value of a profile property."));
+    options.add("T <value>", ki18n("Debian policy compatibility, not used"));
     options.add("!e <cmd>",ki18n("Command to execute"));
     options.add("+[args]",ki18n("Arguments passed to command"));
 }
