From: Debian/Kubuntu Qt/KDE Maintainers <debian-qt-kde@lists.debian.org>
Date: Wed, 13 Apr 2016 12:36:54 +0200
Subject: kubuntu_folderwidget_drawing

===================================================================
---
 plasma/applets/folderview/iconview.cpp | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/plasma/applets/folderview/iconview.cpp b/plasma/applets/folderview/iconview.cpp
index 1ff14af..fc3276a 100644
--- a/plasma/applets/folderview/iconview.cpp
+++ b/plasma/applets/folderview/iconview.cpp
@@ -2588,22 +2588,16 @@ void IconView::startDrag(const QPointF &pos, QWidget *widget)
     pixmap.fill(Qt::transparent);
 
     QStyleOptionViewItemV4 option = viewOptions();
-    // ### We can't draw the items as selected or hovered since Qt doesn't
-    //     use an ARGB window for the drag pixmap.
-    //option.state |= QStyle::State_Selected;
-    option.state &= ~(QStyle::State_Selected | QStyle::State_MouseOver);
+    option.state |= QStyle::State_Selected;
 
     QPainter p(&pixmap);
     foreach (const QModelIndex &index, indexes)
     {
         option.rect = visualRect(index).translated(-boundingRect.topLeft());
-#if 0
-        // ### Reenable this code when Qt uses an ARGB window for the drag pixmap
         if (index == m_hoveredIndex)
             option.state |= QStyle::State_MouseOver;
         else
             option.state &= ~QStyle::State_MouseOver;
-#endif
         paintItem(&p, option, index);
     }
     p.end();
