From: Debian/Kubuntu Qt/KDE Maintainers <debian-qt-kde@lists.debian.org>
Date: Wed, 13 Apr 2016 12:36:54 +0200
Subject: kubuntu_folderview_livecd_directory

===================================================================
---
 plasma/applets/folderview/folderview.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/plasma/applets/folderview/folderview.cpp b/plasma/applets/folderview/folderview.cpp
index 23cac19..6da247e 100644
--- a/plasma/applets/folderview/folderview.cpp
+++ b/plasma/applets/folderview/folderview.cpp
@@ -27,6 +27,7 @@
 #include <QGraphicsSceneDragDropEvent>
 #include <QImageReader>
 #include <QItemSelectionModel>
+#include <QFile>
 
 #include <KAction>
 #include <KAuthorized>
@@ -431,6 +432,10 @@ void FolderView::init()
                 path = QString("desktop:/");
             }
         }
+
+        if (QFile::exists("/usr/share/applications/kde4/oem-config-prepare-kde.desktop") || QFile::exists("/usr/share/applications/kde4/ubiquity-kdeui.desktop")) {
+            path = QString("desktop:/");
+        }
         setUrl(cg.readEntry("url", KUrl(path)));
     }
 
