From a74f99409d463837094c996c19e5f90fe92dea04 Mon Sep 17 00:00:00 2001
From: Pino Toscano <pino@kde.org>
Date: Sat, 18 Jun 2016 08:48:19 +0200
Subject: [PATCH] set cmake_min_req to match kdelibs policy and enable newer
 cmake policies

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b06ba01..2aae22e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,7 @@
 project(KDEBASE_APPS)
 
+cmake_minimum_required(VERSION 2.8.9)
+
 # where to look first for cmake modules, before ${CMAKE_ROOT}/Modules/ is checked
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules )
 
-- 
2.8.1

